torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1049,7 +1063,11 @@ void ProcessGroupNCCL::abortCommsFromMap(
   for (auto& it : ncclCommsMap) {
     auto& devName = it.first;
     auto& ncclComm = it.second;
-
+    at::cuda::OptionalCUDAGuard gpuGuard;
+    at::DeviceIndex deviceIndex = getIndexFromDeviceKey(devName);
```

Comment:
thanks for making getIndexFromDeviceKey safe (above).

but i'm still not sure if `devName` is going to be an int, or could it be a 'cuda:1' type thing?

Reply:
in non p2p comms, it is just an int derived on line 196 just above this function.
deviceKey is generated using the function above on line 196,  device.index() is always an integer: https://www.internalfb.com/code/fbsource/[5335cf5a3ada336055061d4680cc214d21e22ba2]/fbcode/caffe2/c10/core/Device.h?lines=19. For P2P cases, it is in the format of "rank1:rank2"